weaver/cli.py

```diff
@@ -0,0 +1,747 @@
+            inputs = load_file(inputs)
+        if not inputs or not isinstance(inputs, (dict, list)):
+            return OperationResult(False, "No inputs or invalid schema provided.", inputs)
+        if isinstance(inputs, list):  # OLD format provided directly
+            inputs = {"inputs": inputs}
+        # consider possible ambiguity if literal CWL input is named 'inputs'
+        # - if value of 'inputs' is an object, it can collide with 'OGC' schema,
+        #   unless 'value/href' are present or their sub-dict don't have CWL 'class'
+        # - if value of 'inputs' is an array, it can collide with 'OLD' schema,
+        #   unless 'value/href' (and 'id' technically) are present
+        values = inputs.get("inputs", null)
+        if (
+            values is null or
+            values is not null and (
+                (isinstance(values, dict) and get_any_value(values) is null and "class" not in values) or
+                (isinstance(values, list) and all(isinstance(v, dict) and get_any_value(v) is null for v in values))
+            )
+        ):
+            values = cwl2json_input_values(inputs)
+        data = {
+            # NOTE: since sync is not yet properly implemented in Weaver, simulate with monitoring after if requested
+            # FIXME: support 'sync' (https://github.com/crim-ca/weaver/issues/247)
+            "mode": EXECUTE_MODE_ASYNC,
+            "inputs": values,
+            # FIXME: support 'response: raw' (https://github.com/crim-ca/weaver/issues/376)
+            "response": EXECUTE_RESPONSE_DOCUMENT,
+            # FIXME: allow omitting 'outputs' (https://github.com/crim-ca/weaver/issues/375)
+            # FIXME: allow 'transmissionMode: value/reference' selection (https://github.com/crim-ca/weaver/issues/377)
+            "outputs": {}
+        }
+        # FIXME: since (https://github.com/crim-ca/weaver/issues/375) not implemented, auto-populate all the outputs
+        base = self._get_url(url)
+        result = self.describe(process_id, url=base)
+        if not result.success:
+            return OperationResult(False, "Could not obtain process description for execution.",
+                                   body=result.body, headers=result.headers, code=result.code, text=result.text)
+        outputs = result.body.get("outputs")
+        for output_id in outputs:
+            # use 'value' to have all outputs reported in body as 'value/href' rather than 'Link' headers
+            data["outputs"][output_id] = {"transmissionMode": EXECUTE_TRANSMISSION_MODE_VALUE}
+
+        LOGGER.info("Executing [%s] with inputs:\n%s", process_id, _json2text(inputs))
+        path = f"{base}/processes/{process_id}/execution"  # use OGC-API compliant endpoint (not '/jobs')
+        resp = request_extra("POST", path, json=data, headers=self._headers, settings=self._settings)
+        result = self._parse_result(resp)
+        if not monitor or not result.success:
+            return result
```

Comment:
It is not. The `job_reference` here is either the `JobID` or the full URL to it.
From client side, all elements are available as `OperationResult` for all operations, and have to be parsed for chaining steps.
From CLI side, the "main message" followed by JSON body is printed.
This is why the `--monitor` flag is provided in `execute` operation to do the execute+parse+monitor chain.

We could had an option to report more concise details per operation (`--format [json|yaml|simple?]`). 
I personally find more details are useful (especially in cases like deploy or describe).  

@ChaamC @f-PLT 
Could you provide your inputs on this? Would you see a benefit with this kind of option?

Reply:
I prefer having more details too. But a `job_reference` could also include an `OperationResult` object from the execute function for convenience. This would make chaining steps much more easier.

Reply:
I think there is more potential confusion in listing `OperationResult` as a valid input, since it makes sense only if it specifically came from `execute`. Any other operation returning the same object type would be errorneous.

There is also another potential problem. The job URL is obtained by `Location` header returned from `execute`. 
This header is very common since it is used for HTTP redirects. There are chances that Weaver will attempt to do irrelevant operations on a random `Location` it picked from another `OperationResult`.

The proposed `--format` could be `href`. I wouldn't mind adding a specific flag for that case just to output the plain URL string.

Reply:
I will let @ChaamC and @f-PLT give their input on that since they actually used the client. Maybe it's already enough user friendly after all.

Reply:
I think the output is good as it is. The job ID is easy to find if needed for another step, and the details are useful.

Reply:
I guess the addition of a `--format` option could be useful, but as is, I found the CLI and the client very easy to use and the outputs very useful. 

My take on it is, most users will know from the get go if they want to monitor or not the job and the monitor flag will then be used. If monitoring is to be started after, the necessary parsing is simple enough.

Reply:
Will consider it is sufficient for now, and another PR can always be made if such improvements are deemed necessary at some point.